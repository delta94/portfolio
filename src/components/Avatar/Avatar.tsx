import React from 'react';
import { createUseStyles } from 'react-jss';
import { CustomTheme } from '../../theme';

interface AvatarProps {
  url: string;
}

const useStyles = createUseStyles((theme: CustomTheme) => ({
  mainContainer: {
    borderRadius: 50,
    height: 108,
    width: 108,
  },
}));

const UnMemoizedAvatar: React.FC<AvatarProps> = ({ url }) => {
  const classes: Record<'mainContainer', string> = useStyles();

  return <img className={classes.mainContainer} alt='Avatar' src={url} />;
};

const Avatar: React.NamedExoticComponent<AvatarProps> = React.memo(UnMemoizedAvatar);

export { Avatar };
