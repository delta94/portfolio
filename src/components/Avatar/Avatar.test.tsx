import { render } from '@testing-library/react';
import React from 'react';
import { TestContextProvider } from '../../TestContextProvider';
import { Avatar } from './Avatar';

test('Avatar renders correctly', () => {
  const { asFragment } = render(
    <TestContextProvider>
      <Avatar url='mock url' />
    </TestContextProvider>,
  );
  expect(asFragment()).toMatchSnapshot();
});
