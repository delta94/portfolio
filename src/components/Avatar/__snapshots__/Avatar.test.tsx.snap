// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar renders correctly 1`] = `
<DocumentFragment>
  <img
    alt="Avatar"
    class="mainContainer-0-2-1"
    src="mock url"
  />
</DocumentFragment>
`;
